### PR TITLE
Codacy test

### PR DESCRIPTION
**Pull request by egthomas**
This pull request addresses a mix of memory leaks, resource leaks, and other potential errors identified when I was testing Codacy a while back (https://app.codacy.com/project/egthomas/rst/dashboard).  The number of issues listed on their website is buggy (2-3x the actual number) but the individual issues found under the various filters on the "Issues" page should still be accurate when comparing this branch to the develop branch.  

Moving forward I don't plan to do much more with this but am happy to add any future repositories created under the SD organization to Codacy if requested.

### CODE REVIEW

**Comment by egthomas** [comment]
@mts299 pretty much, yeah.

**Comment by mts299** [comment]
@egthomas Do you want to fix the conflicts and then I can take a look at it tomorrow?

File: codebase/superdarn/src.bin/tk/tool/map_addhmb.1.17/map_addhmb.c [hunk 1]
```diff
@@ -74,7 +74,7 @@ int main(int argc,char *argv[])
   int tme;
   int yrsec=0;
   int i,j;
-  int s;
+  int s=0;
```
Comment (mts299) [inline codebase/superdarn/src.bin/tk/tool/map_addhmb.1.17/map_addhmb.c]: Why do you set s = 0 but not i,j on line 76 or others like x on line 89?

File: codebase/superdarn/src.lib/tk/shf.1.10/src/evalvelocity.c [hunk 2]
```diff
@@ -162,12 +162,15 @@ void CnvMapEvalVelocity(int Lmax,double *coef,double *plm,
...
 
   theta=malloc(vptr->num*sizeof(double));
+  if (theta==NULL) return;
   phi=malloc(vptr->num*sizeof(double));
+  if (phi==NULL) {
+    free(theta);
+    return;
+  }
   ex=malloc(vptr->num*sizeof(double));
   ey=malloc(vptr->num*sizeof(double));
```
Comment (mts299) [inline codebase/superdarn/src.lib/tk/shf.1.10/src/evalvelocity.c]: Should we not check if ex and ey are NULL like we did for phi and theta?

File: codebase/superdarn/src.lib/tk/shf.1.10/src/fitvector.c [hunk 3]
```diff
@@ -263,6 +286,10 @@ double CnvMapFitVector(int num,struct CnvMapSHFVec *data,
     free(result);
```
Comment (mts299) [inline codebase/superdarn/src.lib/tk/shf.1.10/src/fitvector.c]: Should we see if w, v, var, and a are NULL as well?

File: codebase/superdarn/src.bin/tk/tool/map_addhmb.1.17/map_addhmb.c (hunk 1 above)
Comment (mts299) [reply codebase/superdarn/src.bin/tk/tool/map_addhmb.1.17/map_addhmb.c]: Funny thing there was a warning for it:
map_addhmb.c:268:11: warning: ‘s’ may be used uninitialized in this function [-Wmaybe-uninitialized]

Maybe we should look at the warnings more closely :p

**Comment by mts299** [comment]
Did Codacy complain about these variables:
map_addmodel.c:696:24: warning: ‘mod_levi[0]’ may be used uninitialized in this function [-Wmaybe-uninitialized]
map_addmodel.c:697:37: warning: ‘mod_levi[1]’ may be used uninitialized in this function [-Wmaybe-uninitialized]
solve_model.c:661:24: warning: ‘mod_levi[0]’ may be used uninitialized in this function [-Wmaybe-uninitialized]
solve_model.c:662:37: warning: ‘mod_levi[1]’ may be used uninitialized in this function [-Wmaybe-uninitialized]

**Comment by egthomas** [comment]
You can see the most recent Travis-CI compilation log for this branch here:  

https://api.travis-ci.org/v3/job/478072179/log.txt

which does not show any "may be used uninitialized" warnings.  Could be a difference of compiler version / settings.

**Comment by mts299** [comment]
Okay, but should it be fixed? 
I think the compiler is complaining because if the `switch` does not happen then `mod_levi` is not set because there is no set default in the switch. 

https://github.com/SuperDARN/rst/blob/a3858d423ef370cda686471bada1d95f5319040b/codebase/superdarn/src.bin/tk/tool/map_addmodel.1.14/map_addmodel.c#L669-L693

**Comment by egthomas** [comment]
It's nearly 5pm on a Friday so this will be the last chance I have to look at this today, but the `interp_coeffs` function is only called by `determine_model` if imod is CS10 or TS18.  So `mod_levi` will always be set by that switch.

**Comment by mts299** [comment]
Okay, I am just pointing out a possible problem. I am not expecting instant fixes. Just trying to test PR's for v4.3 and help out. I can stop now.

**Comment by egthomas** [comment]
@mts299 sorry, I was just trying to let you know I wouldn't be able to respond for a few more days, not that your review comments were not appreciated.  Apologies if it came across that way!

It's worth noting this branch and the `feature/cnvmodel` branch will probably conflict with each other due to similar changes to some of the functions in map_addmodel.

**Comment by mts299** [comment]
I realized this PR might get slammed with some conflicts with other merges :p whoops! But I was hoping to hold off on this one to ensure we didn't introduce more memory leaks with the other fixes.

**Comment by ecbland** [comment]
Is this ready for testing now that the merge conflicts are resolved?

**Comment by egthomas** [comment]
@ecbland yes, I think so.  Not sure what test to recommend since the changes are basically just freeing additional pointers before returning due to an error.

**Comment by ksterne** [comment]
Looks like this still compiles well enough. So merging it in.

**Pull request merged**